src/app/data/http/firestore.service.ts

```diff
@@ -11,41 +12,64 @@ export class FirestoreService {
   constructor(private firestore: AngularFirestore) {
   }
 
+  // todo : snapshot changes used to get also the document id - valuechanges will get only the value
   // get
-  getCollection(collection: string) {
-    return this.firestore.collection(collection).valueChanges();
+  /**
+   * Gets all snapshots of a collection
+   */
+  getCollection(collection: string): Observable<any> {
+    return this.firestore.collection(collection).snapshotChanges();
   }
 
-  getCollectionOrdered(collection: string, orderField: string, orderDirection: OrderByDirection = 'asc') {
+  /**
+   * Gets the last modified data from a collection
+   */
+  getChangedDocuments(collection: string): Observable<any> {
+    return this.firestore.collection(collection).stateChanges();
+  }
+
+  /**
+   * Gets value of a collection ordered by a given value
+   */
+  getCollectionOrdered(collection: string, orderField: string, orderDirection: OrderByDirection = 'asc'): Observable<any> {
     return this.firestore.collection(collection, ref => ref.orderBy('order', orderDirection).limit(10)).valueChanges();
   }
 
-  getDocById(collection: string, id: string) {
-    return this.firestore.collection(collection).doc(id).get();
+  /**
+   * Gets value of a document by document id from a collection
+   */
+  getDocById(collection: string, id: string): Observable<any> {
+    return this.firestore.collection(collection).doc(id).valueChanges();
   }
 
-  getCollectionByWhereClause(collection: string, key, operator, value) {
+  /**
+   * Gets value of a collection where a match is found
+   */
+  getCollectionByWhereClause(collection: string, key, operator, value): Observable<any> {
     return this.firestore.collection(collection, ref => ref.where(key, operator, value)).valueChanges();
   }
 
   // save
-  saveDocumentByAutoId(collection: string, data = null) {
-    return this.firestore.collection(collection).doc(this.firestore.createId()).set(Object.assign({}, data));
-  }
-
-  saveDocumentById(collection: string, id: string, data = null) {
-    return this.firestore.collection(collection).doc(id).set(data);
+  /**
+   * Saves a new document into a collection
+   */
+  saveDocumentByAutoId(collection: string, data = null): Promise<any> {
+    return this.firestore.collection(collection).add(Object.assign({}, data));
```

Comment:
Should we use JSON.parse(JSON.stringigy(data)) instead of Object.assign() to avoid swallow copy? Which is not an issue now, but if we add for some reason a collection to the appointments document it could become one.